src/screens/browse/components/Modals/SourceSettings.tsx

```diff
@@ -75,6 +78,7 @@ const SourceSettingsModal: React.FC<SourceSettingsModal> = ({
     Object.entries(formValues).forEach(([key, value]) => {
       storage.set(key, value);
     });
+    unloadPlugin(pluginId);
```

Comment:
unload plugin on settings change (itll get re-loaded when needed) as this allows re-initialization of plugin settings